### ISSUE TITLE
Prevent iD uploading changes with broken multipolygons

### ISSUE THREAD

**Issue by Gazer75** [openstreetmap/iD#8544]
This is getting out of hand. More and more often I have to fix iD uses breaking relations.
Force iD to check the entire relation and refuse upload where it is broken, or prevent uses from modifying them at all.

**Comment by mmd-osm** [comment]
Can you provide some real examples please, including changeset id, and/or relation ids. Also describe is what ways relations were broken, etc. Otherwise, this issue is not actionable in any way.

**Comment by Gazer75** [comment]
My latest repairs:
https://www.openstreetmap.org/changeset/106477831
https://www.openstreetmap.org/changeset/106359848
https://www.openstreetmap.org/changeset/101197011

Hope this can lead you to the problems.

**Comment by matkoniecz** [comment]
> Also describe is what ways relations were broken, etc.

was missing

--------------------------

and now, in 2026: is it still a problem?

**Comment by Gazer75** [comment]
Still happens from time to time, yes.
Small relations are harder to spot, but I believe I fixed a fairly large wood area that got broken earlier this year. They are easy to spot on the map with a big square of missing green.

Problem is people cut/split lines and don't understand how they are related and with a gap the area is broken.
In the last fix someone had deleted or cut in a bunch of "outer" lines of one or more wood areas. Can't remember exactly why, or if I even understood the reason.

**Comment by danieldegroot2** [comment]
---

Sidenote: as of [iD 2.41.0](https://github.com/openstreetmap/iD/releases/tag/v2.41.0), the iD editor should automatically download missing data when splitting a line if needed. If you have any issues with this feature, please open a new issue.

**Comment by matkoniecz** [comment]
Can you provide some real examples please, including changeset id, and/or relation ids. Also describe is what ways relations were broken, etc. Otherwise, this issue is not actionable in any way.

**Comment by danieldegroot2** [comment]
For reference from earlier 2021 examples
In https://osmcha.org/changesets/106176844 user tried to merge two large wood areas/MPs, deleting the line(way) which divided the two.
The other changesets I can't say directly, they probably had some intersecting geometry where they modified a line in the relation.

Sidenote: a multipolygon with a split outer way should remain a valid multipolygon<sup>([1](https://github.com/openstreetmap/iD/issues/11267#issuecomment-4243256388))</sup>

**Comment by Gazer75** [comment]
> Can you provide some real examples please, including changeset id, and/or relation ids. Also describe is what ways relations were broken, etc. Otherwise, this issue is not actionable in any way.

I was able to find a old changeset from 9 months ago that I remember took a long time. I suddenly noticed most of the grey residential landuse was gone around the area so I investigated. https://www.openstreetmap.org/changeset/173025259
Not sure if you can figure out what exactly got broken and how from my changeset. There are 17 modified relations in there :)
I believe I redid a lot of the ways that are part of the relations as there were broken lines all over the place and some closed polygons overlapping others.

**Comment by danieldegroot2** [comment]
You can see your changes visually at https://osmcha.org/changesets/173025259
If you click on the `+` plus symbol in the left sidebar, open the deleted objects list and scroll down, you can select the deleted ways and see some were from an import and some were touched by other users. One or more ways were touched as recently as 2025 by [peet192](https://resultmaps.neis-one.org/osm-discussion-comments?uid=4337759) who has broken relations several times using iD. (no offense to them) amongst others.

Fyi there are tools to see gaps in relations like Osmose https://osmose.openstreetmap.fr/ (look for 'broken polygons', etc.)
(some Osmose issues are available in iD editor and JOSM editor)

**Comment by Gazer75** [comment]
For some reason OSMCha is not working properly for me. I just get blank pages.

**Comment by matkoniecz** [comment]
may be https://github.com/openstreetmap/iD/issues/12494

**Comment by danieldegroot2** [comment]
@Gazer75 OSMCha has trouble loading changesets with relations. Sometimes it works, most times it does not ;-)

@matkoniecz It is issue with OSMCha, not iD.

**Comment by ojdo** [comment]
Yes, still a problem as of 2026; I recently started participating in a HOTOSM task in Congo, and then just by browsing the nearby region, I stumbled upon multiple broken wood multipolygons. I [asked and got up to speed](https://community.openstreetmap.org/t/how-to-report-mis-rendered-landuse-polygons-e.xample-included/144869) on how to fix those instances over on Discourse. I have 2 relations as examples

* [11986819](https://www.openstreetmap.org/relation/11986819) had a missing outline member. Re-adding it fixed it.
* [11911086](https://www.openstreetmap.org/relation/11911086), but two of its inner members were overlapping, fixed through [changeset 184597249](https://www.openstreetmap.org/changeset/184597249).

I have not looked into the change histories, but I guess these are side effects of such collaborative iD-only mapping efforts. If iD were to have more thorough checks, it would probably also need more JOSM-level validator like error messages and "zoom to problem" helpers for mappers not to frustratingly stop submitting changes.

**Comment by tyrasd** [comment]
> 11986819, 11911086

the breaking edits to those relations seem to have happened using JOSM, so I would say this is not relevant in this case.

Normally, iD tries to prevent breaking multipolygon relations, e.g. by not allowing to delete features as long as they are a member of a relation.

If you notice concrete cases in how iD can create invalid broken multipolygon relations (when starting with a valid one), please report these cases as a new issue (ideally including steps how to reproduce them, if known).
